app/code/community/Ebizmarts/MailChimp/Model/Observer.php

```diff
@@ -338,88 +524,67 @@ public function loadCustomerToQuote(Varien_Event_Observer $observer)
+            $items = $creditMemo->getAllItems();
 
-        return $observer;
-    }
+            foreach ($items as $item) {
+                if ($item->getProductType() == 'bundle' || $item->getProductType() == 'configurable') {
+                    continue;
+                }
 
-    /**
-     * Set the products included in the credit memo to be updated on MailChimp on the next cron job run.
-     *
-     * @param  Varien_Event_Observer $observer
-     * @return Varien_Event_Observer
-     * @throws Exception
-     */
-    public function newCreditMemo(Varien_Event_Observer $observer)
-    {
-        $creditMemo = $observer->getEvent()->getCreditmemo();
-        $order = $creditMemo->getOrder();
-        $mailchimpStoreId = $this->makeHelper()->getMCStoreId($order->getStoreId());
-        $items = $creditMemo->getAllItems();
-        foreach ($items as $item) {
-            if ($item->getProductType() == 'bundle' || $item->getProductType() == 'configurable') {
-                continue;
+                $helper->saveEcommerceSyncData($item->getProductId(), Ebizmarts_MailChimp_Model_Config::IS_PRODUCT, $mailchimpStoreId, null, null, 1, null, null, null, true);
             }
 
-            $this->makeHelper()->saveEcommerceSyncData($item->getProductId(), Ebizmarts_MailChimp_Model_Config::IS_PRODUCT, $mailchimpStoreId, null, null, 1, null, null, null, true);
+            $helper->saveEcommerceSyncData($order->getEntityId(), Ebizmarts_MailChimp_Model_Config::IS_ORDER, $mailchimpStoreId, null, null, 1, null, null, null, true);
         }
-
-        $this->makeHelper()->saveEcommerceSyncData($order->getEntityId(), Ebizmarts_MailChimp_Model_Config::IS_ORDER, $mailchimpStoreId, null, null, 1, null, null, null, true);
         return $observer;
     }
 
     /**
      * Set the products included in the credit memo to be updated on MailChimp on the next cron job run.
      *
-     * @param  Varien_Event_Observer $observer
+     * @param Varien_Event_Observer $observer
      * @return Varien_Event_Observer
-     * @throws Exception
      */
     public function cancelCreditMemo(Varien_Event_Observer $observer)
     {
         $creditMemo = $observer->getEvent()->getCreditmemo();
         $order = $creditMemo->getOrder();
-        $mailchimpStoreId = $this->makeHelper()->getMCStoreId($order->getStoreId());
-        $items = $creditMemo->getAllItems();
-        foreach ($items as $item) {
-            if ($item->getProductType() == 'bundle' || $item->getProductType() == 'configurable') {
-                continue;
+        $storeId = $order->getStoreId();
+        $helper = $this->makeHelper();
+        $ecomEnabled = $helper->isEcomSyncDataEnabled($storeId);
+
+        if ($ecomEnabled) {
+
+            $mailchimpStoreId = $helper->getMCStoreId($storeId);
+
+            $items = $creditMemo->getAllItems();
+            foreach ($items as $item) {
+                if ($item->getProductType() == 'bundle' || $item->getProductType() == 'configurable') {
```

Comment:
use `Mage_Catalog_Model_Product_Type::TYPE_BUNDLE` and `Mage_Catalog_Model_Product_Type::TYPE_CONFIGURABLE`